core/storage/suggestion/gae_models_test.py

```diff
@@ -2994,3 +2994,1639 @@ def test_get_model_association_to_user(self) -> None:
+            last_contribution_date=datetime.datetime.utcnow()
+        ).put()
+        suggestion_models.TranslationSubmitterTotalContributionStatsModel(
+            id='model_3',
+            language_code=self.SUGGESTION_LANGUAGE_CODE,
+            contributor_id=self.USER_ID_1,
+            topic_ids_with_translation_submissions=[
+                'topic1', 'topic2'
+            ],
+            recent_review_outcomes=self.RECENT_REVIEW_OUTCOMES,
+            recent_performance=3,
+            overall_accuracy=1,
+            submitted_translations_count=50,
+            submitted_translation_word_count=(
+                self.SUBMITTED_TRANSLATION_WORD_COUNT),
+            accepted_translations_count=self.ACCEPTED_TRANSLATIONS_COUNT,
+            accepted_translations_without_reviewer_edits_count=(
+                self.ACCEPTED_TRANSLATIONS_WITHOUT_REVIEWER_EDITS_COUNT),
+            accepted_translation_word_count=(
+                self.ACCEPTED_TRANSLATION_WORD_COUNT),
+            rejected_translations_count=self.REJECTED_TRANSLATIONS_COUNT,
+            rejected_translation_word_count=(
+                self.REJECTED_TRANSLATION_WORD_COUNT),
+            first_contribution_date=datetime.datetime.utcnow(),
+            last_contribution_date=datetime.date(2023, 4, 4)
+        ).put()
+        suggestion_models.TranslationSubmitterTotalContributionStatsModel(
+            id='model_4',
+            language_code='hi',
+            contributor_id=self.USER_ID_1,
+            topic_ids_with_translation_submissions=(
+                self.TOPIC_IDS_WITH_TRANSLATION_SUBMISSIONS),
+            recent_review_outcomes=self.RECENT_REVIEW_OUTCOMES,
+            recent_performance=4,
+            overall_accuracy=1,
+            submitted_translations_count=4,
+            submitted_translation_word_count=(
+                self.SUBMITTED_TRANSLATION_WORD_COUNT),
+            accepted_translations_count=self.ACCEPTED_TRANSLATIONS_COUNT,
+            accepted_translations_without_reviewer_edits_count=(
+                self.ACCEPTED_TRANSLATIONS_WITHOUT_REVIEWER_EDITS_COUNT),
+            accepted_translation_word_count=(
+                self.ACCEPTED_TRANSLATION_WORD_COUNT),
+            rejected_translations_count=self.REJECTED_TRANSLATIONS_COUNT,
+            rejected_translation_word_count=(
+                self.REJECTED_TRANSLATION_WORD_COUNT),
+            first_contribution_date=datetime.datetime.utcnow(),
+            last_contribution_date=datetime.date(2023, 2, 4)
+        ).put()
+
+        # Check for decreasing performance(default) sort order.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationSubmitterTotalContributionStatsModel
+            .fetch_page(
+            page_size=2,
+            offset=0,
+            sort_by=None,
+            topic_ids=None,
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_3')
```

Comment:
Check len(sorted_results) and check sorted_results[1].id as well if it exists.

Ditto for other fetch_page() queries, do the test fully.

You should also be creating the data so that each sort produces a different result. You might need to check the second page of results as well for full confirmation. It is OK to make your page size larger when testing sort order, and have a separate test for checking pagination.

Reply:
Updated tests. PTAL again. Thanks.